### PR TITLE
Truffleobject compare

### PR DESCRIPTION
**Pull request by mickjordan**
Some related changes to prior PR

### CODE REVIEW

File: projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMAddressCompareNode.java [hunk 1]
```diff
@@ -51,6 +51,22 @@ public boolean executeI1(TruffleObject val1, TruffleObject val2) {
...
+        }
+
+        @SuppressWarnings("unused")
+        @Specialization(guards = "isCNull(val2)")
+
+        public boolean executeI1(TruffleObject val1, LLVMAddress val2) {
+            return false;
+        }
+
+        public static boolean isCNull(LLVMAddress val) {
```
Comment (mrigger) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMAddressCompareNode.java]: Do we need this guard? Can we not just return false in every case?

Comment (mickjordan) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMAddressCompareNode.java]: I originally just had it as false, but really the only comparison that should be legal is a comparison against NULL. Anything else is at best suspicious, so I thought it better to catch that case and deal with it if an when it arises.

Comment (chrisseaton) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMAddressCompareNode.java]: Does that answer your question @mrigger? Ready to merge?

Comment (mrigger) [inline projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMAddressCompareNode.java]: Yes, I am okay with it.

**Pull request merged**